### PR TITLE
Update to node 18

### LINKED ISSUE #1015: Update to Node 18
Node 16 end-of-life is this [September 11th](https://nodejs.org/en/blog/announcements/nodejs16-eol). 

This repo will need a quick update, to use node 18 (currently 16.x). Potential references to node version:
- GitHub workflows: add at least node 18 in the "matrix" (we may keep 16 for now). 
- package.json "engines.node" version ranges may need updating. e.g. ">=16 <=20 " could work for now
- dockerfiles
- documentation, e.g. README, guides

### PR DESCRIPTION
**Pull request by marcdumais-work**
Also update the required yarn version, that's somewhat related.

Fixes #1015

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -15,15 +15,15 @@ jobs:
...
+        node-version: [18]
 
     steps:
     - name: install dependencies on ubuntu
       if: startsWith(matrix.os,'ubuntu')
       run: |
         sudo apt install -y make gcc pkg-config build-essential libx11-dev libxkbfile-dev
 
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
```
Comment (MatthewKhouzam) [inline .github/workflows/build.yml:26]: should this be v4?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ jobs:
       run: echo "::set-output name=dir::$(yarn cache dir)"
 
     - name: Cache node_modules with yarn
-      uses: actions/cache@v2
+      uses: actions/cache@v3
```
Comment (MatthewKhouzam) [inline .github/workflows/build.yml:38]: v4

File: .github/workflows/publish-next.yml [hunk 3]
```diff
@@ -9,11 +9,11 @@ jobs:
     if: github.ref == 'refs/heads/master' && github.event_name == 'push' && github.repository == 'eclipse-cdt-cloud/theia-trace-extension' 
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
```
Comment (MatthewKhouzam) [inline .github/workflows/publish-next.yml:12]: v4

File: .github/workflows/build.yml (hunk 1 above)
Comment (marcdumais-work) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 2 above)
Comment (marcdumais-work) [reply .github/workflows/build.yml]: done

File: .github/workflows/publish-next.yml (hunk 3 above)
Comment (marcdumais-work) [reply .github/workflows/publish-next.yml]: done

**Review by MatthewKhouzam — APPROVED** [review]

**Pull request merged**